git/git.go

```diff
@@ -30,13 +31,20 @@ type repositoryProvider struct {
 	repository      *git.Repository
 	includePatterns []glob.Glob
 	excludePatterns []glob.Glob
+	fileListToSnap  map[string]bool
```

Comment:
apparently there are not sets in golang, so this is the best next thing:
```go
map[string]bool
```

Reply:
Why can't use an array?

Reply:
I just want to say I had to discover this myself as well in AoC, it's infuriating that such a basic data structure is not part of the standard library.
Had to vent, thank you for coming to my TED talk.

Reply:
@ariel-levy Since I want to be efficient and not run in `O(n*m)` (n- number of lines, m- number of files in list)
this will run at `O(n)`